pkg/querier/blocks_store_queryable.go

```diff
@@ -291,14 +290,7 @@ type blocksStoreQuerier struct {
 // Select implements storage.Querier interface.
 // The bool passed is ignored because the series is always sorted.
 func (q *blocksStoreQuerier) Select(_ bool, sp *storage.SelectHints, matchers ...*labels.Matcher) storage.SeriesSet {
-	set := q.selectSorted(sp, matchers...)
-
-	// We need to wrap the error in order to have Prometheus returning a 5xx error.
-	if err := set.Err(); err != nil && !errors.Is(err, context.Canceled) && !errors.Is(err, context.DeadlineExceeded) {
```

Comment:
@pracucci: I'm not sure about this change... here if error was context.Canceled or DeadlineExceeded, it was reported to prom/api directly. Previously it meant 422, now it means 500. Not sure which one is better. For canceled -- if we can assume it was cancelled by user -- I think we should still return 422. What about DeadlineExceeded?

Reply:
`context.Canceled` should definitely by a 422. `context.DeadlineExceeded` is probably more fair to be considered as 500, but I'm open to discuss it.

Reply:
Let's keep `context.DeadlineExceeded` as 500 for now (I don't see any "deadline exceeded" errors returned by our queriers or query-frontends in last week).